protocol/http.h

```diff
@@ -54,34 +57,90 @@ struct Response {
 
 class Http {
 public:
-    template<class SocketType>
-    static Response get(SocketType &&socket, uri::Uri const &uri) {
+    static Response get(auto &&socket, uri::Uri const &uri) {
+        using namespace std::string_view_literals;
+
         if (socket.connect(uri.authority.host, Http::use_port(uri) ? uri.authority.port : uri.scheme)) {
             socket.write(Http::create_get_request(uri));
-            auto data = socket.read_until("\r\n");
+            auto data = socket.read_until("\r\n"sv);
             if (data.empty()) {
                 return {Error::Unresolved};
             }
             auto status_line = Http::parse_status_line(data.substr(0, data.size() - 2));
             if (!status_line) {
                 return {Error::InvalidResponse};
             }
-            data = socket.read_until("\r\n\r\n");
+            data = socket.read_until("\r\n\r\n"sv);
             if (data.empty()) {
                 return {Error::InvalidResponse};
             }
             auto headers = Http::parse_headers(data.substr(0, data.size() - 4));
             if (headers.size() == 0) {
                 return {Error::InvalidResponse};
             }
-            data = socket.read_all();
+            auto encoding = headers.get("transfer-encoding"sv);
+            if (encoding.has_value() && *encoding == "chunked"sv) {
```

Comment:
This can be written as `if (encoding != "chunked"sv) {`. `std::optional`'s comparison operators internally check if it has a value, and if not, returns false when compared to something with a value.